networkx/algorithms/clique.py

```diff
@@ -18,19 +18,21 @@
 #    All rights reserved.
 #    BSD license.
 from collections import deque
-from itertools import chain, islice
+from itertools import chain
+from itertools import combinations
+from itertools import islice
 try:
     from itertools import ifilter as filter
 except ImportError:
     pass
 import networkx
-from networkx.utils.decorators import *
+from ..utils import not_implemented_for
```

Comment:
@jfinkels absolute import is a good idea here given the fact that `utils` will probably stay where it is,  and also that the rest of the codebase uses absolute import.

Also, I think it is a bad idea to assume here that utils will always be two levels up, because that could change with the rearranging of sub-packages. Moreover, its imported into the top level namespace for a reason? Correct me if I'm wrong.